samples/cpp/benchmark_app/main.cpp

```diff
@@ -1175,48 +1175,55 @@ int main(int argc, char* argv[]) {
-                const auto& data = inputsData.at(inputName)[0];
-                inferRequest->set_tensor(inputName, data);
+        // conditional warmup based on FLAGS_no_warmup
+        if (!FLAGS_no_warmup) {
+            // warming up - out of scope
+            inferRequest = inferRequestsQueue.get_idle_request();
+            if (!inferRequest) {
+                OPENVINO_THROW("No idle Infer Requests!");
             }
+            if (!inferenceOnly) {
+                auto inputs = app_inputs_info[0];
 
-            if (useGpuMem) {
-                auto outputTensors =
-                    ::gpu::get_remote_output_tensors(compiledModel, inferRequest->get_output_cl_buffer());
-                for (auto& output : compiledModel.outputs()) {
-                    inferRequest->set_tensor(output.get_any_name(), outputTensors[output.get_any_name()]);
+                for (auto& item : inputs) {
+                    auto inputName = item.first;
+                    const auto& data = inputsData.at(inputName)[0];
+                    inferRequest->set_tensor(inputName, data);
+                }
+
+                if (useGpuMem) {
+                    auto outputTensors =
+                        ::gpu::get_remote_output_tensors(compiledModel, inferRequest->get_output_cl_buffer());
+                    for (auto& output : compiledModel.outputs()) {
+                        inferRequest->set_tensor(output.get_any_name(), outputTensors[output.get_any_name()]);
+                    }
                 }
             }
-        }
 
-        if (FLAGS_api == "sync") {
-            inferRequest->infer();
-        } else {
-            inferRequest->start_async();
-        }
+            if (FLAGS_api == "sync") {
+                inferRequest->infer();
+            } else {
+                inferRequest->start_async();
+            }
 
-        inferRequestsQueue.wait_all();
+            inferRequestsQueue.wait_all();
 
-        auto duration_ms = inferRequestsQueue.get_latencies()[0];
-        slog::info << "First inference took " << double_to_string(duration_ms) << " ms" << slog::endl;
+            auto duration_ms = inferRequestsQueue.get_latencies()[0];
+            slog::info << "First inference took " << double_to_string(duration_ms) << " ms" << slog::endl;
 
-        if (statistics) {
-            statistics->add_parameters(
-                StatisticsReport::Category::EXECUTION_RESULTS,
-                {StatisticsVariant("first inference time (ms)", "first_inference_time", duration_ms)});
+            if (statistics) {
+                statistics->add_parameters(
+                    StatisticsReport::Category::EXECUTION_RESULTS,
+                    {StatisticsVariant("first inference time (ms)", "first_inference_time", duration_ms)});
+            }
+            inferRequestsQueue.reset_times();
+        } else {
+            slog::info << "Skipping warmup inference due to -no_warmup flag" << slog::endl;
         }
```

Comment:
```suggestion
        }
```
There are concerns that benchmark_app is already too verbose. No need to print to confirm that it's going to do exactly what the user requested

Reply:
I genuinely think this could stay and isn't contributing to the verbosity of benchmark_app, as it confirms the user has requested correctly with -no+warmup. It takes exactly one line, instead of "First inference took X seconds"